metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/update/ESBulkProcessor.java

```diff
@@ -109,6 +122,76 @@ public Optional<BulkByScrollResponse> deleteByQuery(QueryBuilder queryBuilder, b
         return Optional.empty();
     }
 
+    public Optional<Cancellable> reindex(QueryBuilder queryBuilder, String index, int batchSize, @Nullable TimeValue timeout) {
+        final long startTime = System.currentTimeMillis();
+        String destIndex = index + "_" + startTime;
+        ReindexRequest reindexRequest = new ReindexRequest()
```

Comment:
One of the things we do in the SystemUpdate job is check for existing reindexes running for the requested index using the task api. We don't want multiple calls to this to result in multiple reindexes running.

Reply:
This truncation is a bit different then reindexing (for a settings/mapping change). In the case where we are truncating, I think we could create the new index and update the alias to the index right away. Any active append-only writes would be directed there and then a purely async process to copy over the subset of the old index happens in the background. By swapping the order for the timeseries index, we can eliminate the importance of using the cancel-able to make sure the alias swap is done later. Definite makes sense to prevent duplicate re-indexing in any case.